tests/ApiProblemTest.php

```diff
@@ -42,6 +52,27 @@ public function testComplexExtraProperty() : void
         $this->assertEquals('Zim', $problem['irken']['invader']);
     }
 
+    public function testSimpleJsonCompileWithJsonException() : void
+    {
+        $this->expectException(JsonEncodeException::class);
+        $this->expectExceptionCode(\JSON_ERROR_UTF8);
+        $this->expectExceptionMessage('Malformed UTF-8 characters, possibly incorrectly encoded');
+
+        // an invalid string value
+        $problem = new ApiProblem(\hex2bin('58efa99d4e19ff4e93efd93f7afb10a5'), 'URI');
```

Comment:
Out of curiosity, what is this random string you're feeding in? :smile:

Reply:
It is a GUID, I used it in another project about LDAP and I knew it fails when json tries to serialize it. :)